tests/Functional/ContentEventRenderTest.php

```diff
@@ -183,42 +167,511 @@ public function testEventRegistrationDateDescription(): void {
+    $node->set('oe_event_venue', [$venue_entity])->save();
+    $this->drupalGet($node->toUrl());
+
+    $field_list_expected_values = [
+      'items' => [
+        [
+          'label' => 'Where',
+          'body' => "event_venue\n  Address event_venue, 1001 Brussels, Belgium",
+        ], [
+          'label' => 'When',
+          'body' => "Wednesday 19 February 2020, 01:00\n to Friday 28 February 2020, 01:00",
+        ], [
+          'label' => 'Languages',
+          'body' => 'Estonian, French',
+        ],
+      ],
+    ];
+    // @todo: should be removed when PHP 7.2 support will be finished.
+    if (version_compare(PHP_VERSION, '7.3') < 0) {
+      $field_list_expected_values['items'][0]['body'] = "event_venue\n\n  Address event_venue, 1001 Brussels, Belgium";
+    }
+    $field_list_assert->assertPattern($field_list_expected_values, $practical_list_content->getOuterHtml());
+
+    $icons_text_expected_values['items'][2] = [
+      'icon' => 'location',
+      'text' => 'Brussels, Belgium',
+    ];
+    $icons_text_assert->assertPattern($icons_text_expected_values, $details_list_content->getOuterHtml());
+
+    // Assert "Internal organiser" field.
+    $node->set('oe_event_organiser_is_internal', TRUE);
+    $node->set('oe_event_organiser_internal', 'http://publications.europa.eu/resource/authority/corporate-body/AASM');
+    $node->save();
+    $this->drupalGet($node->toUrl());
+
+    $field_list_expected_values['items'][3] = [
+      'label' => 'Organiser',
+      'body' => 'Associated African States and Madagascar',
+    ];
+    $field_list_assert->assertPattern($field_list_expected_values, $practical_list_content->getOuterHtml());
+
+    // Assert "Organiser name" field.
+    $node->set('oe_event_organiser_is_internal', FALSE);
+    $node->set('oe_event_organiser_name', 'External organiser')->save();
+    $this->drupalGet($node->toUrl());
+
+    $field_list_expected_values['items'][3] = [
+      'label' => 'Organiser',
+      'body' => 'External organiser',
+    ];
+    $field_list_assert->assertPattern($field_list_expected_values, $practical_list_content->getOuterHtml());
+
+    // Assert "Online type", "Online link", "Online time" fields.
+    $online_start_date = (clone $static_time)->modify('+ 1 months');
+    $online_end_date = (clone $static_time)->modify('+ 2 months');
+    $node->set('oe_event_online_type', 'facebook');
+    $node->set('oe_event_online_link', [
+      'uri' => 'http://www.example.com/online_link',
+      'title' => 'Link to online event',
+    ]);
+    // @todo: "Online description" isn't shown on the page. Why do we need it?
+    $node->set('oe_event_online_description', 'Online event description');
+    $node->set('oe_event_online_dates', [
+      'value' => $online_start_date->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT),
+      'end_value' => $online_end_date->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT),
+    ])->save();
+    $this->drupalGet($node->toUrl());
```

Comment:
When testing all the different messages, that depends on the current time, we shouldn't touch the node at all, we only have to modify the time, via the datetime testing, and run cron, as we do on Behat:

![image](https://user-images.githubusercontent.com/153362/116205041-a485ae00-a73d-11eb-96db-6be754750bb4.png)

This is important since, if we modify the node and reload the page, then cache is invalidated by modifying the actual node, so we are not testing that time related cache tags (set in the extra fields) are working correctly.

Everything related to time testing should never involve modifying the node dates, but it should always consist of the following steps:

- Change frozen time to desired one
- Run cron (this will invalidate the cache)
- Reload the page
- Run assertions

Reply:
Done